core/templates/dev/head/pages/exploration_editor/exploration_objective_editor_directive.html

```diff
@@ -0,0 +1,24 @@
+<form name="objectiveForm">
+  <p ng-class="{'has-error': objectiveForm.$dirty && (!explorationObjectiveService.displayed || explorationObjectiveService.displayed.length < 15 || explorationObjectiveService.displayed.length > 100)}">
+    <label for="explorationObjective" ng-style="<[::formStyle]>"><[::labelText]></label>
+    <input id="explorationObjective" type="text"
+          class="form-control protractor-test-exploration-objective-input"
+          ng-model="explorationObjectiveService.displayed"
+          ng-blur="onInputFieldBlur()" placeholder="Learn how to..." ng-trim="false">
+    <span class="help-block help-block-directive" ng-show="objectiveForm.$dirty && explorationObjectiveService.displayed.length < 15" style="font-size: smaller">
```

Comment:
The **help-block-directive** is a CSS class that I created specially to solve the issue. It has the same attributes and values as the **help-block** class but, with a different bottom-margin value, to reduce the excessive spacing.